jvector-base/src/main/java/io/github/jbellis/jvector/graph/GraphIndexBuilder.java

```diff
@@ -263,13 +263,14 @@ private void reconnectOrphanedNodes() {
 
                 // no unused candidate found -- search for more neighbors and try again
                 neighbors = searchPathNeighbors.get(node);
-                if (neighbors == null) {
+                // run search again if neighbors is empty or if every neighbor is already in connection targets
+                if (neighbors == null || isSubset(neighbors, connectionTargets)) {
```

Comment:
half-baked thought: would it be cleaner to mutate the cached neighbors as we add connections?

Reply:
let's leave it alone, then